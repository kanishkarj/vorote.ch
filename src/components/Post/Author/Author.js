// @flow strict
import React from 'react';
import { getContactHref } from '../../../utils';
import styled from 'styled-components';
import { useSiteMetadata } from '../../../hooks';
import { margin } from '../../../utils/get-margin';
import { lineHeight } from '../../../utils/get-line-height';
import { mediaUp } from '../../../utils/get-media-query';

const StyledAuthor = styled.div`
  border-top: 1px solid var(--color-gray500);
  padding-top: 20px;
  ${lineHeight(1)}
  ${margin(1, 0)}
  ${mediaUp.sm`
    margin-left: auto;
    margin-right: auto;
  `}
`;

const TwitterLink = styled.a`
  display: block;
  text-decoration: underline;
`;

const Author = () => {
  const { author } = useSiteMetadata();

  return (
    <StyledAuthor>
      <p>
        {author.bio}
        <TwitterLink
          href={getContactHref('twitter', author.contacts.twitter)}
          rel="noopener noreferrer"
          target="_blank"
        >
          <strong>{author.name}</strong> on Twitter
        </TwitterLink>
      </p>
    </StyledAuthor>
  );
};

export default Author;
