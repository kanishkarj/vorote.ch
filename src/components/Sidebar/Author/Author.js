// @flow strict
import React from 'react';
import { withPrefix, Link } from 'gatsby';
import styled from 'styled-components';
import { margin, marginBottom } from '../../../utils/get-margin';
import { lineHeight } from '../../../utils/get-line-height';

const Photo = styled.img`
  display: inline-block;
  margin-bottom: 0;
  border-radius: 50%;
  background-clip: padding-box;
`;

const IndexTitle = styled.h1`
  font-size: ${(props) => props.theme.fontSizes.base * 1.125}px;
  font-weight: ${(props) => props.theme.fontWeights.heading};
  ${lineHeight(1.125)}
  ${margin(0.5, 0)}
`;

const Title = styled.h2`
  font-size: ${(props) => props.theme.fontSizes.base * 1.125}px;
  font-weight: ${(props) => props.theme.fontWeights.heading};
  ${lineHeight(1.125)}
  ${margin(0.5, 0)}
`;

const TitleLink = styled(Link)`
  color: var(--color-text);
  &:hover,
  &:focus {
    color: var(--color-text);
  }
`;

const Subtitle = styled.p`
  color: var(--color-gray500);
  ${lineHeight(1)}
  ${marginBottom(1)}
`;

type Props = {
  author: {
    name: string,
    bio: string,
    photo: string
  },
  isIndex: ?boolean
};

const Author = ({ author, isIndex }: Props) => (
  <div>
    <Link to="/">
      <Photo
        src={withPrefix(author.photo)}
        width="75"
        height="75"
        alt={author.name}
      />
    </Link>

    {isIndex === true ? (
      <IndexTitle>
        <TitleLink to="/">{author.name}</TitleLink>
      </IndexTitle>
    ) : (
        <Title>
          <TitleLink to="/">{author.name}</TitleLink>
        </Title>
    )}
    <Subtitle>{author.bio}</Subtitle>
  </div>
);

export default Author;
