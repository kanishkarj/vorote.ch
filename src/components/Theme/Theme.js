import React from 'react';
import { ThemeProvider } from 'styled-components';

const theme = {
  fonts: {
    body: '-apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen-Sans, Ubuntu, Cantarell, "Helvetica Neue", sans-serif;',
    heading: '-apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen-Sans, Ubuntu, Cantarell, "Helvetica Neue", sans-serif;',
    monospace: 'Menlo, monospace',
  },
  fontWeights: {
    body: 400,
    heading: 600,
    bold: 700,
  },
  fontSizes: {
    base: 16,
    small: 14,
  },
  lineHeight: 1.625,
  spacing: Math.round(16 * 1.625),
  buttons: {
    height: 35,
    borderRadius: 20,
  },
  breakpoints: {
    xs: 0,
    sm: 576,
    md: 768,
    lg: 992,
    xl: 1200
  }
};

const Theme = ({ children }) => (
  <ThemeProvider theme={theme}>
    {children}
  </ThemeProvider>
);

export default Theme;
