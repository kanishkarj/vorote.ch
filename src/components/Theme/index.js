// @flow strict
export { default } from './Theme';
