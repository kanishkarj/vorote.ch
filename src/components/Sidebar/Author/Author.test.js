// @flow strict
import React from 'react';
import renderer from 'react-test-renderer';
import Author from './Author';
import 'jest-styled-components';
import Theme from '../../Theme';

describe('Author', () => {
  const props = {
    author: {
      name: 'test',
      photo: '/photo.jpg',
      bio: 'test'
    },
    isIndex: false
  };

  it('renders correctly', () => {
    const tree = renderer.create(
      <Theme>
        <Author {...props} />
      </Theme>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
