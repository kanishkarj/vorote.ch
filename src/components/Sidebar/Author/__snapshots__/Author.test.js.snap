// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Author renders correctly 1`] = `
.c0 {
  display: inline-block;
  margin-bottom: 0;
  border-radius: 50%;
  background-clip: padding-box;
}

.c1 {
  font-size: 18px;
  font-weight: 600;
  line-height: 1.828125em;
  margin: 13px 0px;
}

.c2 {
  color: var(--color-text);
}

.c2:hover,
.c2:focus {
  color: var(--color-text);
}

.c3 {
  color: var(--color-gray500);
  line-height: 1.625em;
  margin-bottom: 26px;
}

<div>
  <a
    href="/"
  >
    <img
      alt="test"
      className="c0"
      height="75"
      src="/photo.jpg"
      width="75"
    />
  </a>
  <h2
    className="c1"
  >
    <a
      className="c2"
      href="/"
    >
      test
    </a>
  </h2>
  <p
    className="c3"
  >
    test
  </p>
</div>
`;
